zilliqa/src/db.rs

```diff
@@ -1,19 +1,181 @@
-use eth_trie::DB;
+use std::path::Path;
+
+use anyhow::Result;
 use sled::Tree;
 
+use crate::{
+    crypto::Hash,
+    message::{Block, BlockHeader},
+    state::{Address, SignedTransaction, TransactionReceipt},
+};
+
+#[derive(Debug)]
+pub struct Db {
+    root: sled::Db,
+    block_header: Tree,
+    block: Tree,
+    canonical_block_number: Tree,
+    transaction: Tree,
+    transaction_receipt: Tree,
+    /// An index of address to a list of transaction hashes, for which this address appeared somewhere in the
+    /// transaction trace. The list of transations is ordered by execution order.
+    touched_address_index: Tree,
+}
+
+macro_rules! get_and_insert_methods {
```

Comment:
Not too sure I am a big fan of this macro as it just reduces a little duplication, but if it is going to be in here could it add a commment?